### PR TITLE
GEN-1645 - refact(ProductReviews): move dialog to separate component

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Refactoring: extract `ReviewsDialog` into its own component.

Can be tested [here](https://hedvig-dot-com-git-gen-1645-refactreviews-dialog-hedvig.vercel.app/se/forsakringar/olycksfallsforsakring/av)

## Justify why they are needed

Makes `ProductReviews` component cleaner.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3793?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3789** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3789?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3790** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3790?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3791** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3791?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3792** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3792?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #3793** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3793?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3793?utm_source=stack-comment).

File: apps/store/src/components/ProductReviews/ReviewsDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  onSelectedTabChange,
+  selectedScore,
+  onSelectedScoreChange,
+  tooltipText,
+}: Props) => {
+  const { t } = useTranslation('common')
+
+  return (
+    <Dialog.Root>
+      <Dialog.Trigger asChild>
```
Comment (robinandeer) [inline apps/store/src/components/ProductReviews/ReviewsDialog.tsx:41]: nit: asChild={true}

**Review by robinandeer — APPROVED** [review]
nice refactor

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jan 2, 10:30 AM**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3793).

**Pull request merged**